Incorrect behaviour of File-based Session Property Manager

<!--- Provide a general summary of the issue in the Title above -->
<!--- Look through existing open and closed issues to see if someone has reported the issue before -->

File-based session property manager throws a null pointer exception if no catalog session property is provided in any JSON entry

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Map.entrySet()" because "map" is null
at com.google.common.collect.ImmutableMap.copyOf(ImmutableMap.java:688)
at com.facebook.presto.session.SessionMatchSpec.<init>(SessionMatchSpec.java:74)
...
```

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Presto version used: 0.295-SNAPSHOT
* Storage (HDFS/S3/GCS..): N.A
* Data source and connector used: N.A.
* Deployment (Cloud or On-prem): N.A.

## Expected Behavior
<!--- Tell us what should happen -->
Presto should allow one of the sessionProperties or catalogSessionProperties arguments to be null

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
If catalogSessionProperties is null and sessionProperties is non-null:
```
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Map.entrySet()" because "map" is null
at com.google.common.collect.ImmutableMap.copyOf(ImmutableMap.java:688)
at com.facebook.presto.session.SessionMatchSpec.<init>(SessionMatchSpec.java:74)
...
```

If sessionProperties is null and catalogSessionProperties is non-null:
```
java.lang.IllegalArgumentException: sessionProperties is null
```

--- a/presto-file-session-property-manager/src/test/java/com/facebook/presto/session/file/TestFileSessionPropertyManager.java
+++ b/presto-file-session-property-manager/src/test/java/com/facebook/presto/session/file/TestFileSessionPropertyManager.java
@@ -19,12 +19,14 @@ import com.facebook.presto.session.AbstractTestSessionPropertyManager;
 import com.facebook.presto.session.SessionMatchSpec;
 import com.facebook.presto.spi.session.SessionPropertyConfigurationManager;
 import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
 
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Optional;
 
 import static com.facebook.presto.session.file.FileSessionPropertyManager.CODEC;
 import static org.testng.Assert.assertEquals;
@@ -58,5 +60,61 @@ public class TestFileSessionPropertyManager
             assertEquals(propertyConfiguration.systemPropertyOverrides, overrideProperties);
             assertEquals(manager.getCatalogSessionProperties(CONTEXT), catalogProperties);
         }
+    }
+
+    @Test
+    public void testNullSessionProperties()
+            throws IOException
+    {
+        ImmutableMap<String, Map<String, String>> catalogProperties = ImmutableMap.of("CATALOG", ImmutableMap.of("PROPERTY", "VALUE"));
+        SessionMatchSpec spec = new SessionMatchSpec(
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                null,
+                catalogProperties);
+
+        assertProperties(ImmutableMap.of(), ImmutableMap.of(), catalogProperties, spec);
+    }
+
+    @Test
+    public void testNullCatalogSessionProperties()
+            throws IOException
+    {
+        Map<String, String> properties = ImmutableMap.of("PROPERTY1", "VALUE1", "PROPERTY2", "VALUE2");
+        SessionMatchSpec spec = new SessionMatchSpec(
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                properties,
+                null);
+
+        assertProperties(properties, spec);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Either sessionProperties or catalogSessionProperties must be provided")
+    public void testNullBothSessionProperties()
+            throws IOException
+    {
+        SessionMatchSpec spec = new SessionMatchSpec(
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                null,
+                null);
+
+        assertProperties(ImmutableMap.of(), spec);
     }
 }

--- a/presto-session-property-managers-common/src/main/java/com/facebook/presto/session/SessionMatchSpec.java
+++ b/presto-session-property-managers-common/src/main/java/com/facebook/presto/session/SessionMatchSpec.java
@@ -69,9 +69,12 @@ public class SessionMatchSpec
         this.resourceGroupRegex = requireNonNull(resourceGroupRegex, "resourceGroupRegex is null");
         this.clientInfoRegex = requireNonNull(clientInfoRegex, "clientInfoRegex is null");
         this.overrideSessionProperties = requireNonNull(overrideSessionProperties, "overrideSessionProperties is null");
-        requireNonNull(sessionProperties, "sessionProperties is null");
-        this.sessionProperties = ImmutableMap.copyOf(sessionProperties);
-        this.catalogSessionProperties = ImmutableMap.copyOf(catalogSessionProperties);
+
+        checkArgument(sessionProperties != null || catalogSessionProperties != null,
+                "Either sessionProperties or catalogSessionProperties must be provided");
+
+        this.sessionProperties = ImmutableMap.copyOf(Optional.ofNullable(sessionProperties).orElse(ImmutableMap.of()));
+        this.catalogSessionProperties = ImmutableMap.copyOf(Optional.ofNullable(catalogSessionProperties).orElse(ImmutableMap.of()));
     }
 
     public <T> Map<String, T> match(Map<String, T> object, SessionConfigurationContext context)

